datamap/example/src/main/java/org/apache/carbondata/datamap/minmax/AbstractMinMaxDataMapWriter.java

```diff
@@ -0,0 +1,248 @@
+        !FileFactory.mkdirs(dataMapPath, FileFactory.getFileType(dataMapPath))) {
+      throw new IOException("Failed to create directory " + dataMapPath);
+    }
+
+    try {
+      currentIndexFile = MinMaxIndexDataMap.getIndexFile(dataMapPath,
+          MinMaxIndexHolder.MINMAX_INDEX_PREFFIX + indexColumns.size());
+      FileFactory.createNewFile(currentIndexFile, FileFactory.getFileType(currentIndexFile));
+      currentIndexFileOutStream = FileFactory.getDataOutputStream(currentIndexFile,
+          FileFactory.getFileType(currentIndexFile));
+    } catch (IOException e) {
+      CarbonUtil.closeStreams(currentIndexFileOutStream);
+      LOGGER.error("Failed to init datamap index file", e);
+      throw e;
+    }
+  }
+
+  protected void resetBlockletLevelMinMax() {
+    for (int i = 0; i < indexColumns.size(); i++) {
+      indexColumnMinMaxCollectors[i].getPageStats().clear();
+    }
+  }
+
+  @Override
+  public void onBlockStart(String blockId) {
+  }
+
+  @Override
+  public void onBlockEnd(String blockId) {
+  }
+
+  @Override public void onBlockletStart(int blockletId) {
+  }
+
+  @Override public void onBlockletEnd(int blockletId) {
+    flushMinMaxIndexFile();
+    currentBlockletId++;
+  }
+
+  @Override
+  public void onPageAdded(int blockletId, int pageId, int pageSize, ColumnPage[] pages) {
+    // as an example, we don't use page-level min-max generated by native carbondata here, we get
+    // the min-max by comparing each row
+    for (int rowId = 0; rowId < pageSize; rowId++) {
+      for (int colIdx = 0; colIdx < indexColumns.size(); colIdx++) {
+        Object originValue = pages[colIdx].getData(rowId);
+        updateBlockletMinMax(colIdx, originValue);
+      }
+    }
+  }
+
+  protected void updateBlockletMinMax(int indexColIdx, Object value) {
+    if (null == value) {
+      indexColumnMinMaxCollectors[indexColIdx].updateNull(0);
+      return;
+    }
+
+    CarbonColumn indexCol = indexColumns.get(indexColIdx);
+    DataType dataType = indexCol.getDataType();
+    if (indexCol.isMeasure()
+        || (indexCol.isDimension()
+        && DataTypeUtil.isPrimitiveColumn(dataType)
+        && !indexCol.hasEncoding(Encoding.DICTIONARY)
+        && !indexCol.hasEncoding(Encoding.DIRECT_DICTIONARY))) {
+      if (DataTypes.BOOLEAN == dataType) {
+        indexColumnMinMaxCollectors[indexColIdx].update(
+            BooleanConvert.boolean2Byte((boolean) value));
+      } else if (DataTypes.SHORT == dataType) {
+        indexColumnMinMaxCollectors[indexColIdx].update((short) value);
+      } else if (DataTypes.INT == dataType) {
+        indexColumnMinMaxCollectors[indexColIdx].update((int) value);
+      } else if (DataTypes.LONG == dataType || DataTypes.TIMESTAMP == dataType) {
+        indexColumnMinMaxCollectors[indexColIdx].update((long) value);
+      } else if (DataTypes.DOUBLE == dataType) {
+        indexColumnMinMaxCollectors[indexColIdx].update((double) value);
+      } else if (DataTypes.isDecimal(dataType)) {
```

Comment:
Please handle for DataTypes.FLOAT datatype also

Reply:
OK